src/network/networkd-manager.c

```diff
@@ -1444,6 +1446,12 @@ int manager_new(Manager **ret, sd_event *event) {
 
         (void) routing_policy_load_rules(m->state_file, &m->rules_saved);
 
+        // Router lifetime can be set with netlink interface since kernel >= 4.5
+        // so for the supported kernel we dont need to expire routes in userspace
+        route_expiration_condition = condition_new(CONDITION_KERNEL_VERSION, ">= 4.5", false, false);
```

Comment:
So, either the result of `condition_new` should be used for oom, and the result of `condition_test` should be check for error, or maybe, which I think is a nicer option, just allocate this statically and assert:
```C
{
         static const Condition c = { .type = CONDITION_KERNEL_VERSION, .parameter = ">= 4.5");

         r = condition_test(route_expiration_condition);
         assert(r >= 0);
         m->route_expiration_kernel = !!r;
}
```